sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/SensorContextUtils.java

```diff
@@ -0,0 +1,269 @@
+ */
+package org.sonar.plugin.typescript;
+
+import com.google.gson.JsonElement;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.issue.NewIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+public class SensorContextUtils {
+
+  private static final Logger LOG = Loggers.get(ContextualSensor.class);
+
+  private SensorContextUtils() {
+  }
+
+  static void saveIssues(SensorContext sensorContext, Issue[] issues, TypeScriptRules typeScriptRules) {
+    FileSystem fs = sensorContext.fileSystem();
+    for (Issue issue : issues) {
+      InputFile inputFile = fs.inputFile(fs.predicates().hasAbsolutePath(issue.name));
+      if (inputFile != null) {
+        saveIssue(sensorContext, typeScriptRules, issue, inputFile);
+      }
+    }
+  }
+
+  public static List<RuleToExecute> convertToRulesToExecute(TypeScriptRules typeScriptRules) {
+    List<RuleToExecute> rulesToExecute = new ArrayList<>();
+
+    typeScriptRules.forEach(rule -> {
+      if(rule.isEnabled()) {
+        rulesToExecute.add(new RuleToExecute(rule.tsLintKey(), rule.configuration()));
+      }
+    });
+
+    return rulesToExecute;
+  }
+
+  static void saveIssuePocToRemove(SensorContext sensorContext, TypeScriptRules typeScriptRules, Issue issue, InputFile inputFile) {
+    RuleKey ruleKey = RuleKey.of("typescript", "S1764");
+    NewIssue newIssue = sensorContext.newIssue().forRule(ruleKey);
+    NewIssueLocation location = newIssue.newLocation();
+    location.on(inputFile);
+    location.message(issue.failure);
+
+    // semicolon rule
+    if (ruleKey.rule().equals("S1438")) {
+      location.at(inputFile.selectLine(issue.startPosition.line + 1));
+
+    } else if (!TypeScriptRules.FILE_LEVEL_RULES.contains(ruleKey.rule())) {
+      location.at(inputFile.newRange(
+        issue.startPosition.line + 1,
+        issue.startPosition.character,
+        issue.endPosition.line + 1,
+        issue.endPosition.character));
+    }
+
+    newIssue.at(location);
+
+    // there is not secondaryLocations for issues coming from tslint rules
+    if (issue.secondaryLocations != null) {
+      for (SecondaryLocation secondaryLocation : issue.secondaryLocations) {
+        NewIssueLocation newSecondaryLocation = newIssue.newLocation().on(inputFile);
+        setSecondaryLocation(newSecondaryLocation, secondaryLocation, inputFile);
+        newIssue.addLocation(newSecondaryLocation);
+      }
+    }
+
+    if (issue.cost != null) {
+      newIssue.gap(issue.cost);
+    }
+    LOG.info("Saving issue " + issue + "\n  as >>>> " + newIssue); // TODO Remove this
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this trailing comment on the previous empty line. ([squid:TrailingCommentCheck](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonarlint-integration-poc&issues=AWDgL5jfbBlXo5JPxg68&open=AWDgL5jfbBlXo5JPxg68)

[ISSUE_KEY]: AWDgL5jfbBlXo5JPxg68